api/src/com/cloud/network/guru/NetworkGuruAdditionalFunctions.java

```diff
@@ -0,0 +1,12 @@
+package com.cloud.network.guru;
+
+import java.util.Map;
+
+public interface NetworkGuruAdditionalFunctions {
```

Comment:
I agree. This was the "best" solution I found, maybe for my lack CS architecture knowledge. This was the problem that I faced:
I needed to call this functions from NetworkOrchestrator. However, from NetworkOrchestrator I could access NetworkGuru interface, so I decided not to include this methods in NetworkGuru interface, implementing them blank in every guru except in NiciraNvpGuestNetworkGuru. I decided to create this interface, only implemented by NiciraNvpGuestNetworkGuru just to avoid doing that.

In NetworkOrchestrator, line 682 I made a call to one of this functions. How can I improve this solution?